src/OmniSharp.MSBuild/ProjectManager.cs

```diff
@@ -483,22 +483,31 @@ private void UpdateReferences(Project project, ImmutableArray<string> referenceP
                 if (!File.Exists(referencePath))
                 {
                     _logger.LogWarning($"Unable to resolve assembly '{referencePath}'");
+                    continue;
                 }
-                else
+
+                // There is no need to explicitly add assembly to workspace when the assembly is produced by a project reference.
+                // Doing so actually can cause /codecheck request to return errors like below for types in the referenced project if it is for example signed:
+                // The type 'TestClass' exists in both 'SignedLib, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' and 'ClassLibrary1, Version=1.0.0.0, Culture=neutral, PublicKeyToken=a5d85a5baa39a6a6'
+                if (TryFindLoadedProjectReferenceWithTargetPath(referencePath, projectReferencePaths, project.Name, out ProjectFileInfo projectReferenceWithTarget))
```

Comment:
Should we just build up a `HashSet<string>` of project target paths ahead of time to avoid searching the list of project references for each metadata reference?

Reply:
Given that typically the list of project references is relatively short IMO it is acceptable here to sacrifice a little bit of perf for avoiding managing an additional piece of workspace state, particularly for the cases when the list of project is updated.
Please advise if you still would prefer me to look into adding this optimization.

Reply:
No need to make the change. Just asking.